confiture-web-app/src/components/ui/FileList.vue

```diff
@@ -12,12 +13,17 @@ export interface FileListFile {
   key: string;
 }
 
-const { files } = defineProps<{
+const { files, isInModal, onDelete } = defineProps<{
   files: FileListFile[];
   readonly?: boolean;
+  isInModal?: boolean;
+  onDelete?: (flFile: FileListFile, triggerButton?: EventTarget | null) => void;
```

Comment:
Pour pouvoir `await` un appel à `onDelete` il faut que la fonction retourne un `Promise`


```suggestion
  onDelete?: (flFile: FileListFile, triggerButton?: EventTarget | null) => Promise<void>;
```